### PR TITLE
Arnavic/add-eslint

### PR DESCRIPTION
**Pull request by arnavic**
Adding ES Lint configs from Varnish 1.0 + Preserving Material UI ESLint rules within this new package as well 
Tested on the varnish package in this repo with the local eslint package route and is yielding the same results as installing the eslint-config from Varnish 1.0, so should be good to go.

### CODE REVIEW

**Comment by jonborchardt** [comment]
sorry for not being clearer...
i dont think we should preserve material mui linting rules
this package will be used on all other demos, but we wont use it on this app

so

1- please dont include the material mui eslint rules
2- dont use this package in this project
3- we need documentation added in an 'extensions' section 
4- please name the folder and package varnish-eslint-config

**Comment by arnavic** [comment]
> sorry for not being clearer... i dont think we should preserve material mui linting rules this package will be used on all other demos, but we wont use it on this app
> 
> so
> 
> 1- please dont include the material mui eslint rules 2- dont use this package in this project 3- we need documentation added in an 'extensions' section 4- please name the folder and package varnish-eslint-config

got it! i will remove the mui-lint rules, and i haven't used this in Varnish in this PR anyways i just did that as a testing thing to sanity check that this linting package was working as expected :) will do the needful.

File: docs/data/material/pagesVarnish.js [hunk 1]
```diff
@@ -25,6 +25,8 @@ module.exports = [
   {
     pathname: '/material-ui/varnish/extensions',
     subheader: 'EXTENSIONS',
-    children: [],
+    children: [
+      { pathname: '/material-ui/varnish/varnish-extensions', title: 'Varnish Extensions' },
```
Comment (jonborchardt) [inline docs/data/material/pagesVarnish.js:29]: i think there will be more extensions... so please rename this file to `varnish-eslint-config`

File: docs/data/material/varnish/varnish-extensions/varnish-extensions.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+---
+title: AI2 Varnish Extensions
```
Comment (jonborchardt) [inline docs/data/material/varnish/varnish-extensions/varnish-extensions.md:2]:
```suggestion
title: varnish-eslint-config
```

File: docs/data/material/varnish/varnish-extensions/varnish-extensions.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+githubLabel: 'varnish: varnish-extensions'
```
Comment (jonborchardt) [inline docs/data/material/varnish/varnish-extensions/varnish-extensions.md:3]:
```suggestion
githubLabel: 'varnish: varnish-eslint-config'
```

File: docs/data/material/varnish/varnish-extensions/varnish-extensions.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+# Extensions
```
Comment (jonborchardt) [inline docs/data/material/varnish/varnish-extensions/varnish-extensions.md:6]: please make this md about varnish-eslint-config, not about extensions in general

File: packages/eslint-config-varnish/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+   tsconfig.json
+   EOF
+   ```
+
+3. Create a config file, and configure `eslint` to use this package as a base:
+
+   ```bash
+   cat <<EOF > .eslintrc.js
+   module.exports = {
+       extends: [ "@allenai/eslint-config-varnish2" ]
```
Comment (jonborchardt) [inline packages/eslint-config-varnish/README.md:51]:
```suggestion
       extends: [ "@allenai/varnish-eslint-config" ]
```

**Review by jonborchardt — COMMENTED** [review]
all references should be to varnish-eslint-config
and the md file should be about varnish-eslint-config not extensions...

File: docs/data/material/varnish/varnish-extensions/varnish-extensions.md (hunk 4 above)
Comment (arnavic) [reply docs/data/material/varnish/varnish-extensions/varnish-extensions.md:6]: sure will do!

File: packages/eslint-config-varnish/README.md (hunk 5 above)
Comment (arnavic) [reply packages/eslint-config-varnish/README.md:51]: thanks for catching, missed in the various file changes..... :D

File: docs/data/material/varnish/varnish-eslint-config/varnish-eslint-config.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+---
+title: varnish-eslint-config
+githubLabel: 'varnish: varnish-eslint-config'
+---
+
+# Extensions
```
Comment (jonborchardt) [inline docs/data/material/varnish/varnish-eslint-config/varnish-eslint-config.md:6]:
```suggestion
# Varnish ESLint Config
```

File: docs/data/material/varnish/varnish-eslint-config/varnish-eslint-config.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+   ```
+   # See what's wrong
+   ~ yarn lint
+
+   # Reformat and fix things
+   ~ yarn lint:fix
+   ```
+
+For more information about Varnish ESLint and further customization options, visit the [documentation](https://github.com/allenai/varnish-mui/tree/main/packages/eslint-config-varnish/README.md).
```
Comment (jonborchardt) [inline docs/data/material/varnish/varnish-eslint-config/varnish-eslint-config.md:41]:
```suggestion
For more information about Varnish ESLint and further customization options, visit the [documentation](https://github.com/allenai/varnish-mui/tree/main/packages/varnish-eslint-config/README.md).
```

File: packages/eslint-config-varnish/README.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+   ```bash
+   # See what's wrong
+   ~ yarn lint
+
+   # Reformat and fix things
+   ~ yarn lint:fix
+   ```
+
+## Publishing
```
Comment (arnavic) [inline packages/eslint-config-varnish/README.md:75]: @jonborchardt - should i delete this for now while we don't have publishing instructions?

File: packages/eslint-config-varnish/README.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+For more information about publishing and using shared `eslint` configuration
+definitions, see [this documentation](https://eslint.org/docs/user-guide/configuring).
+
+## Usage
+
+1. Install it and it's peer dependencies:
+
+   ```shell
+   ~ yarn add @allenai/eslint-config-varnish \
```
Comment (jonborchardt) [inline packages/eslint-config-varnish/README.md:18]:
```suggestion
   ~ yarn add @allenai/varnish-eslint-config \
```

File: packages/eslint-config-varnish/index.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
+/**
+ * This file includes `eslint` settings that are distributed via the
+ * `@allenai/eslint-config-varnish` package and shared amongst many AI2
```
Comment (jonborchardt) [inline packages/eslint-config-varnish/index.js:3]:
```suggestion
 * `@allenai/varnish-eslint-config` package and shared amongst many AI2
```

File: packages/eslint-config-varnish/package.json [hunk 11]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (jonborchardt) [inline packages/eslint-config-varnish/package.json:1]: please rename the folder to packages/varnish-eslint-config

**Review by jonborchardt — COMMENTED** [review]
more edits

File: packages/eslint-config-varnish/README.md (hunk 8 above)
Comment (jonborchardt) [reply packages/eslint-config-varnish/README.md:75]: no, i will add it there soon

File: packages/eslint-config-varnish/package.json (hunk 11 above)
Comment (arnavic) [reply packages/eslint-config-varnish/package.json:1]: yep on it, sorry i was pushing edits in batches so the commit suggestions and local branch didn't diverge much! let me do a final rename push

Comment (jonborchardt) [reply packages/eslint-config-varnish/package.json]: no problem!

Comment (arnavic) [reply packages/eslint-config-varnish/package.json]: done! hoping i caught all of them :D

Comment (jonborchardt) [reply packages/eslint-config-varnish/package.json]: looking :-P

**Review by jonborchardt — APPROVED** [review]
ty, i know this was annoying!

**Pull request merged**